### PR TITLE
Implemented `WrappingPathway` in `sympy.physics.mechanics`

### PR DESCRIPTION
**Pull request by brocksam**
#### References to other Issues or PRs

Forms part of the CZI biomechanics work program described in https://github.com/sympy/sympy/issues/24240. Extends #24914 and #25068.

#### Brief description of what is fixed or changed

This PR introduces a new pathway class: `WrappingPathway`. This is a further concrete class of `PathwayBase` that interacts with concrete classes to `GeometryBase` to create a pathway that wraps around a geometry. An example application for biomechanical modelling is creating a pathway for a musculotendon that produces a geodetic wrapping around a solid cylinder.

I have not added any documentation beyond docstrings, nor have I added the docstrings to the online documentation yet. For the reasoning above this would likely need significant rewriting after any API changes. Documentation will be contributed in a future PR when the module is moved to `sympy/physics/mechanics/pathway.py`.

#### Other comments

I have specifically left the release notes entry empty. These will be added in the PR that moves the module to `sympy/physics/mechanics/pathway.py`.

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by brocksam** [comment]
SymEngine tests are failing due to a difference between how SymPy and SymEngine simplify expressions. Specifically instances where symbols with assumptions are used; I'm using `Symbol('r', positive=True)` for instances where it is nonsensical to have a negative value associated with a symbol, e.g. a radius. Using assumptions here leads to nice automatic simplification in SymPy, but in SymEngine it leaves terms like `sqrt(r**2)` hanging around everywhere instead of just cancelling to `r`.

**Comment by oscarbenjamin** [comment]
> The results with SymEngine are still "correct" in these cases, SymEngine just doesn't cancel terms as expected, even with `simplify`.

Are the expressions involved absurdly large or is it possible to show here what an example looks like?

Perhaps something more canonical like `expand` could be used (`simplify` is not well defined).

**Comment by moorepants** [comment]
I think we should remove symengine from physics.vector/mechanics. It is fundamentally a problem because we can't/won't fix issues in symengine.

**Comment by moorepants** [comment]
I also have zero evidence anyone uses it.

**Comment by brocksam** [comment]
> I think we should remove symengine from physics.vector/mechanics. It is fundamentally a problem because we can't/won't fix issues in symengine.

I think I'm also in favour of this. The fact of the matter is that both @TJStienstra and I seem to spend a large amount of time per PR ensuring that the SymEngine tests pass, which: (1) often makes the library code less clear and less readable; (2) skip writing tests representative of real world use cases because they're harder to get passing for both SymPy and SymEngine simultaneously; and (3) wastes time.

**Comment by moorepants** [comment]
Then let's remove it. For this PR just skip tests for symengine or XFAIL.

I think we could remove symengine backend for physics.vec/mech without deprecation because it does not change results or API, it only has to do with performance. But let's discuss that in a new PR to remove it.

**Comment by brocksam** [comment]
> Then let's remove it. For this PR just skip tests for symengine or XFAIL.

I've added skips, all CI jobs are now passing.

> But let's discuss that in a new PR to remove it.

I've opened #25401 for this.

File: sympy/physics/mechanics/tests/test_pathway.py [hunk 1]
```diff
@@ -147,3 +187,372 @@ def test_3D_pathway_compute_loads(self) -> None:
...
+
+    @pytest.fixture(autouse=True)
+    def _wrapping_pathway_fixture(self) -> None:
+        self.pA = Point('pA')
+        self.pB = Point('pB')
+        self.r = Symbol('r', positive=True)
+        self.pO = Point('pO')
+        self.N = ReferenceFrame('N')
+        self.ax = self.N.z
+        self.sphere = Sphere(self.r, self.pO)
```
Comment (moorepants) [inline sympy/physics/mechanics/tests/test_pathway.py:205]: `.sphere` doesn't seem to be used (or maybe that is hidden somewhere).

Comment (moorepants) [reply sympy/physics/mechanics/tests/test_pathway.py:205]: ah, this is just some setup code

File: sympy/physics/mechanics/_pathway.py [hunk 2]
```diff
@@ -268,3 +275,208 @@ def compute_loads(self, force: ExprType) -> list[LoadBase]:
...
+        >>> from sympy.physics.mechanics import Point, ReferenceFrame
+        >>> from sympy.physics.mechanics._geometry import Cylinder
+        >>> N = ReferenceFrame('N')
+        >>> r = Symbol('r', positive=True)
+        >>> pO = Point('pO')
+        >>> cylinder = Cylinder(r, pO, N.z)
+
+        Create the pathway of the actuator using the ``WrappingPathway`` class,
+        defined to span between two points ``pA`` and ``pB``. Both points lie
+        on the surface of the cylinder and the location of ``pB`` is defined
```
Comment (moorepants) [inline sympy/physics/mechanics/_pathway.py:430]: What guarantee's that points pA and pB lie on the surface of the cylinder?

File: sympy/physics/mechanics/_pathway.py [hunk 3]
```diff
@@ -268,3 +275,208 @@ def compute_loads(self, force: ExprType) -> list[LoadBase]:
...
+        >>> from sympy.physics.mechanics import dynamicsymbols
+        >>> from sympy.physics.mechanics._pathway import WrappingPathway
+        >>> q = dynamicsymbols('q')
+        >>> pA = Point('pA')
+        >>> pB = Point('pB')
+        >>> pA.set_pos(pO, r * N.x)
+        >>> pB.set_pos(pO, r * (cos(q) * N.x + sin(q) * N.y))
+        >>> pB.pos_from(pA)
+        (r*cos(q(t)) - r)*N.x + r*sin(q(t))*N.y
+        >>> pathway = WrappingPathway(pA, pB, cylinder)
```
Comment (moorepants) [inline sympy/physics/mechanics/_pathway.py:443]: From this example it seems that points pA and pB are intended to *always* lie on the cylinder. If that is the case, then why does the user need to specify the radial locations of these points?

File: sympy/physics/mechanics/_pathway.py [hunk 4]
```diff
@@ -268,3 +275,208 @@ def compute_loads(self, force: ExprType) -> list[LoadBase]:
             Force(self.attachments[-1], force * relative_position / self.length),
         ]
         return loads
+
+
+class WrappingPathway(PathwayBase):
+    """Pathway that wraps a geometry object.
+
+    Explanation
+    ===========
```
Comment (moorepants) [inline sympy/physics/mechanics/_pathway.py:284]: This explanation is pretty abstract. A figure showing examples of what this all means would greatly improve it.

**Comment by moorepants** [comment]
I'm just going to merge this and will review it like I did the last one when making it public.

**Pull request merged**